Modules/Core/Transform/test/itkMatrixOffsetTransformBaseGTest.cxx

```diff
@@ -0,0 +1,67 @@
+/*=========================================================================
+ *
+ *  Copyright NumFOCUS
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+
+// First include the header file to be tested:
+#include "itkMatrixOffsetTransformBase.h"
+
+#include <gtest/gtest.h>
+#include <random>
+
+
+namespace
+{
+itk::OptimizerParameters<double>
+GenerateRandomParameters(const unsigned size)
+{
+  itk::OptimizerParameters<double> result(size);
+
+  if (size > 0)
+  {
+    std::mt19937                     randomNumberEngine;
+    std::uniform_real_distribution<> randomDistribution;
+
+    for (auto & parameter : result)
+    {
+      parameter = randomDistribution(randomNumberEngine);
+    }
+  }
+  return result;
+}
+
+
+template <unsigned int NDimensions>
+void
+Assert_SetFixedParameters_throws_when_size_is_less_than_NDimensions()
+{
+  for (unsigned int size{}; size < NDimensions; ++size)
```

Comment:
`unsigned int size{}` is less readable than `unsigned int size=0`. Is there any advantage of initializing with `{}`?

Reply:
@lassoan In this specific case, `T var{}` and `T var=0` are basically equivalent. The expression `T var=0` converts an `int` (zero) to the specified type `T`, and when type `T` is `unsigned int`, that's just fine, of course. Zero performance overhead, no rounding errors, no overflow. But in general, when `T` is a user-defined type, `T var=0` may be significantly slower than `T var{}`, because `T var=0` may call a converting constructor of `T`, whereas `T var{}` may just call the default-constructor. 

So in general, I prefer to avoid conversion, when C++11 `{}` initialization can do the job. Conversion may (in cases other than this specific one) involve rounding errors, overflow, and a performance penalty.

Would it be OK to you if I leave the line of code like this? Or do you want me to change it?

Reply:
`unsigned size = 0u` doesn't have performance penalty. I agree with Andras that `=0` is clearer than `{}` initialization.